Source/options.js

```diff
@@ -418,7 +418,7 @@ $(function() {
       * as well as check if the object data has fields other than
       * activation or not.
       **/
-      if (!data || JSON.stringify(data) == '{}' || !data.savedPatterns) {
+      if (!data || JSON.stringify(data) == '{}' || !data.savedPatterns || typeof data.savedPatterns=='undefined') {
```

Comment:
I don't see how this change would help. it seems to me that whenever "typeof data.savedPatterns=='undefined' " would return true, "!data.savedPatterns" would also return true. Therefore, it seems useless to add this extra expression.

But please correct me if I am wrong.